model_compression_toolkit/xquant/common/tensorboard_utils.py

```diff
@@ -0,0 +1,113 @@
+
+    def get_graph_for_tensorboard_display(self,
+                                          quantized_model: Any,
+                                          similarity_metrics: Dict[str, Any],
+                                          repr_dataset: Callable) -> Graph:
+        """
+        Get the graph for Tensorboard display. The framework-specific implementations
+        (like KerasTensorboardUtils and PytorchTensorboardUtils) should implement this
+        as it differs between them when combining the similarity metrics into the graph.
+
+        Args:
+            quantized_model (Any): The quantized model.
+            similarity_metrics (Dict[str, Any]): Metrics for model similarity.
+            repr_dataset (Callable): Representative dataset function.
+
+        Returns:
+            Graph: The generated graph for Tensorboard display.
+        """
+        Logger.get_logger().critical("This method should be implemented by the framework-specific TensorboardUtils.")
+
+    def add_histograms_to_tensorboard(self,
+                                      model: Any,
+                                      repr_dataset: Callable):
+        """
+        Add histograms to Tensorboard after collecting them on the float folded model using
+        the representative dataset.
+
+        Args:
+            model (Any): The model for which to add histograms.
+            repr_dataset (Callable): The representative dataset to use during statistics collection.
+        """
+        graph = self.model_folding_utils.create_float_folded_graph(model, repr_dataset)
+        mi = ModelCollector(graph, self.fw_impl, self.fw_info)
+        for _data in tqdm(repr_dataset(), desc="Collecting Histograms"):
+            mi.infer(_data)
+        self.tb_writer.add_histograms(graph, "")  # Do not use tag since this is the only graph that is added.
+
+    def add_graph_to_tensorboard(self,
+                                 quantized_model: Any,
+                                 similarity_metrics: Dict[str, Any],
+                                 repr_dataset: Callable):
+        """
+        Add a graph to Tensorboard. The graph represents the quantized graph
+        with the similarity metrics that were measured in different nodes.
+
+        Args:
+            quantized_model (Any): The quantized model.
+            similarity_metrics (Dict[str, Any]): The similarity metrics that were collected.
+            repr_dataset (Callable): Representative dataset to use (if needed, like in pytorch case).
+        """
+        # Generate the quantized graph with similarity metrics.
+        tb_graph = self.get_graph_for_tensorboard_display(quantized_model=quantized_model,
+                                                          similarity_metrics=similarity_metrics,
+                                                          repr_dataset=repr_dataset)
+
+        self.tb_writer.add_graph(tb_graph, "")  # Do not use tag since this is the only graph that is added.
```

Comment:
use const value for the default ""?
there is no case that in the future we want to add other graphs? maybe give it some default graph tag name